contentcuration/contentcuration/static/js/administration/router.js

```diff
@@ -0,0 +1,293 @@
+        "channels(/filter/:filter)(/sort/:key-:order)(/search/:search)(/p:page)(/:pagesize-per-page)":     "channels",    // #channels
+    },
+	execute(callback, args, name){
+		let page = args[4] ? Number(args[4]) : 1
+		let newRouteParams = {
+			name: name,
+			filter: args[0],
+			sortKey: args[1],
+			sortOrder: args[2],
+			search: args[3],
+			page: page,
+			pageSize: Number(args[5]),
+		}
+		let sameTabNewParams = this.currentRouteParams.name === newRouteParams.name &&
+								!_.isEqual(this.currentRouteParams, newRouteParams)
+
+		let switchTabNewParams = this.currentRouteParams.name !== newRouteParams.name &&
+								!_.isEqual(this.routeParamsCache[newRouteParams.name], newRouteParams)
+		if ( 
+			// check to see if we need to reload
+			sameTabNewParams || switchTabNewParams
+		){
+			// if so, execute the route
+			if (callback) callback.apply(this, args);
+		}
+
+		// update the current state
+		this.currentRouteParams = newRouteParams
+		this.routeParamsCache[name] = this.currentRouteParams
+
+		// make sure the correct tab is showing, even if we aren't loading anything
+		$('a.btn.'+name).tab('show')
+		return false
+	},
+	getRoute({name, filter, sortKey, sortOrder, search, page, pageSize}){
+		return name +
+				(filter ? "/filter/" + filter : "") +
+				(	
+					(sortKey || sortOrder) ? 
+					"/sort/" + (sortKey ? sortKey : GET_DEFAULT_ROUTES()[name]['sortKey']) + "-" +
+					(sortOrder ? sortOrder : "ascending")
+					: ""
+				) +
+				(search ? "/search/" + search : "") +
+				(page ? "/p" + page : "") +
+				(pageSize ? "/" + pageSize + "-per-page" : "")
+	},
+	gotoRouteForParams(params){
+		if (!params.name) {
+			// update existing params with incoming ones
+			let currentRouteParams = Object.assign({}, this.currentRouteParams)
+			params = Object.assign(currentRouteParams, params)
+		}
+		let route = this.getRoute(params)
+		this.navigate(route, {trigger: true})
+	},
+	gotoPage(page){
+		this.gotoRouteForParams({page: page})
+	},
+	gotoNextPage(){
+		this.gotoPage(this.currentRouteParams.page+1)
+	},
+	gotoPreviousPage(){
+		this.gotoPage(this.currentRouteParams.page-1)
+	},
+	initialize: function(){
+		var AdministrationView = require("./views");
+		this.admin_view = new AdministrationView.AdminView ({
+			el: $("#admin-container"),
+			router: this,
+		});
+		this.collections = {}
+		this.currentRouteParams = GET_DEFAULT_ROUTES()['channels']
+		this.routeParamsCache = GET_DEFAULT_ROUTES()
+		let router = this;
+		$('.nav-tabs a').click(function(e){
+			e.stopImmediatePropagation()
+			let routeParams = router.routeParamsCache[e.currentTarget.attributes['data-href'].value]
+			router.gotoRouteForParams(routeParams)
+		})
+		window.current_user = new Models.UserModel(window.user);
+	},
+	getSelected(opts) {
+		for (let k in opts){
+			if (opts[k].selected === true) {
+				return k
+			}
+		}
+	},
+	updateCollectionStateFromParams(collection, filter, sortKey, order, search, page = 1, pageSize){
```

Comment:
This function translates state into the relevant backbone collection object.  The backbone collection is responsible for interfacing with the API.